src/ranged.cpp

```diff
@@ -1147,32 +1147,39 @@ static int draw_turret_aim( const player &p, WINDOW *w, int line_number, const t
     return line_number;
 }
 
-// TODO: Shunt redundant drawing code elsewhere
-std::vector<tripoint> game::pl_target_ui( target_mode mode, item *relevant, int range, const itype *ammo,
-                                          const target_callback &on_mode_change,
-                                          const target_callback &on_ammo_change )
+std::vector<tripoint> target_handler::target_ui( player &pc, const targeting_data &args ) {
+    return target_ui( pc, args.mode, args.relevant, args.range,
+                      args.ammo, args.on_mode_change, args.on_ammo_change );
+}
+
+// @todo: Shunt redundant drawing code elsewhere
+std::vector<tripoint> target_handler::target_ui( player &pc, target_mode mode,
+                                                 item *relevant, int range, const itype *ammo,
+                                                 const target_callback &on_mode_change,
+                                                 const target_callback &on_ammo_change )
 {
+    // @todo: this should return a reference to a static vector which is cleared on each call.
```

Comment:
I don't think that would help much. Modern compilers generally optimize the return value and turn it into a move operation.